HSLdevcom/jore4-ui
# Basic stop search sorting and paging

### Add memoize function implementation with a single item cache
Lodash does provide a memoization function, but that has an infinite cache size, thus by default it causes a memory leak.

Inspired by https://github.com/alexreardon/memoize-one, but a simplified version of it, with a very basic functionality. Could be replaced by memoize-one in future, if a more robust implementation is needed.

By default uses Object.is to compare the function parameters with, but allows for alternative comparators too.


### Update Tiamat & regen GraphQL files


### Add some margin to LoadingWrapper


### StopSearch: Add 'searching...' subtext to LoadingStopRow


### StopSearch: Use the new Paging component
* Add paging to StopSearchUrlState.
* Pass down paging info into result stop result list.
* Actual paging still handled UI side. (Backend paging also needs backend side sorting)
* Reset paging, when submitting new filters.


### StopSearch: Add initial URL state bits for sorting
Add types, default values and serialization bits needed to implement proper sorting of the results.


### StopSearch: Add SortResultsBy component


### StopSearch: Add ResultCountHeader component


### StopSearch: Add CountAndSortingRow component


### StopSearch: Implement backend sorting and paging on basic stop search

- - -

### Testing:
1. Go to http://localhost:3300/stop-registry
2. Search for Stops by Label or Address
3. Results should be sortable by Label, Name & Address.
4. Pagination should work 


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/HSLdevcom/jore4-ui/943)
